### ISSUE TITLE
Split up the ffi CI job

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#2782]
The ffi CI job performs a lot of loosely related tasks that mostly don't leverage the previous one's build artifacts, and the ffi job is quickly becoming our slowest job. This could be mitigated by splitting the job apart more.

CC @Manishearth @echeran

**Comment by Manishearth** [comment]
Yeah we definitely should split it up, perhaps as a second wasm job. Or just move it ot the full-datagen job.

**Comment by sffc** [comment]
I went ahead and put gn in its own job, but we should still work on the rest of the ffi job.

**Comment by sffc** [comment]
I think we might actually have 3 places that do full datagen:

1. The full-datagen job
2. The ffi job when it tests JS
3. The wasm-demo job

**Comment by sffc** [comment]
Alternatively, this problem might largely go away if we can leverage a high-quality working cache solution for dependency crates, so we don't need to re-build heavy dependencies all the time.

**Comment by sffc** [comment]
In CI, we should make sure that we run datagen with both the oldest supported data and the newest supported data.

**Comment by sffc** [comment]
Also, we should be testing FFI on other platforms.
